jadx-core/src/test/java/jadx/tests/functional/StringUtilsTest.java

```diff
@@ -50,4 +56,133 @@ public void testCharUnescape() {
 	private void checkCharUnescape(char input, String result) {
 		assertThat(stringUtils.unescapeChar(input), is('\'' + result + '\''));
 	}
+
+	@Test
+	public void testEscape() {
+		Assert.assertEquals("_", StringUtils.escape(","));
+		Assert.assertEquals("A", StringUtils.escape("["));
+		Assert.assertEquals("", StringUtils.escape("*"));
+		Assert.assertEquals("foo", StringUtils.escape("foo"));
+	}
+
+	@Test
+	public void testEscapeXML() {
+		Assert.assertEquals("2", StringUtils.escapeXML("2"));
+		Assert.assertEquals("&apos;", StringUtils.escapeXML("\'"));
+		Assert.assertEquals("&lt;&amp;", StringUtils.escapeXML("<&"));
+		Assert.assertEquals("\\0", StringUtils.escapeXML("\u0000"));
+		Assert.assertEquals("&quot;", StringUtils.escapeXML("\""));
+		Assert.assertEquals("\\\\", StringUtils.escapeXML("\\"));
+		Assert.assertEquals("&gt;", StringUtils.escapeXML(">"));
+		Assert.assertEquals("&amp;", StringUtils.escapeXML("&"));
+	}
+
+	@Test
+	public void testEscapeResValue() {
+		Assert.assertEquals("(", StringUtils.escapeResValue("("));
+		Assert.assertEquals("&amp;", StringUtils.escapeResValue("&"));
+	}
+
+	@Test
+	public void testEscapeResStrValue() {
+		Assert.assertEquals("/", StringUtils.escapeResStrValue("/"));
+		Assert.assertEquals("\\\'", StringUtils.escapeResStrValue("\'"));
+		Assert.assertEquals("\\'", StringUtils.escapeResStrValue("\'"));
+		Assert.assertEquals("2", StringUtils.escapeResStrValue("2"));
+		Assert.assertEquals("\\\"", StringUtils.escapeResStrValue("\""));
+	}
+
+	@Test
+	public void testEscapeWhiteSpaceChar1()
```

Comment:
Can be refactored further.. same as below
`Assert.assertEquals("\\r", methodUnderTest.invoke(null, '\r'));`